MAUI/DataGrid/columns.md

```diff
@@ -0,0 +1,205 @@
+</tr>
+<tr>
+<td> <code>SmartReset</code> </td>
+<td> Retains the columns defined explicitly in application level and the columns with <code>MappingName</code> identical to the properties in the new data source. Creates columns newly for all the other properties in the data source.<br/> Similarly it retains the sorting of the columns that are defined explicitly in application level and the columns with <code>MappingName</code> identical to the properties in new data source. </td>
+</tr>
+</table>
+
+### Customize automatically generated columns
+
+The auto generated column can be customized by handling the [SfDataGrid.AutoGeneratingColumn](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataGrid.SfDataGrid.html#Syncfusion_Maui_DataGrid_SfDataGrid_AutoGeneratingColumn) event. The event will be invoked when the column is auto-generated.
+
+The [DataGridAutoGeneratingColumnEventArgs](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataGrid.DataGridAutoGeneratingColumnEventArgs.html) object contains the following properties,
+
+* **Column**: This property returns the created column which can be customized.
+* **Cancel**: This property cancels the column creation.
+
+You can skip generating a column by handling the `SfDataGrid.AutoGeneratingColumn` event as shown as follows,
+
+{% tabs %}
+{% highlight c# %}
+this.sfDataGrid.AutoGeneratingColumn += SfDataGrid_AutoGeneratingColumn;
+
+private void SfDataGrid_AutoGeneratingColumn(object sender, DataGridAutoGeneratingColumnEventArgs e)
+{
+	if(e.Column.MappingName == "ShipCountry")
+	{
+		e.Cancel = true;
+	}
+}
+{% endhighlight %}
+{% endtabs %}
+
+Formatting for the auto generated columns can be applied as follows:
+
+{% tabs %}
+{% highlight c# %}
+private void SfDataGrid_AutoGeneratingColumn(object sender, DataGridAutoGeneratingColumnEventArgs e)
+{
+    if (e.Column.MappingName == "Freight")
+    {
+        e.Column.Format = "C";
+    }
+    else if (e.Column.MappingName == "ShippingDate")
+    {
+        e.Column.Format = "MMMM dd";
+    }
+}
+{% endhighlight %}
+{% endtabs %}
+
+## Manually generate columns
+
+The `SfDataGrid` allows to define the columns manually by adding the [DataGridColumn](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataGrid.DataGridColumn.html) objects to the [SfDataGrid.Columns](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataGrid.SfDataGrid.html#Syncfusion_Maui_DataGrid_SfDataGrid_Columns) collection. If you want to show only the manually defined columns in the view, you can achieve it by setting the `SfDataGrid.AutoGenerateColumnsMode` property to `None`.

```

Comment:
The `SfDataGrid` allows to define the columns manually by adding the [DataGridColumn](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataGrid.DataGridColumn.html) objects to the [SfDataGrid.Columns](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataGrid.SfDataGrid.html#Syncfusion_Maui_DataGrid_SfDataGrid_Columns) collection. If you want to show only the manually defined columns in the view, you can achieve that by setting the `SfDataGrid.AutoGenerateColumnsMode` property to `None`.

Reply:
Modified.